src/pages/nft/class/_classId/_nftId.vue

```diff
@@ -318,31 +338,33 @@ export default {
         }));
     },
   },
-  asyncData({ query }) {
+  async asyncData({ route, query, store, redirect, error }) {
     const { action } = query;
-    return { action };
-  },
-  async fetch({ route, store, redirect, error }) {
-    const { classId } = route.params;
+    const { classId, nftId } = route.params;
     const { referrer } = route.query;
     if (referrer) {
       redirect({
         name: 'nft-class-classId-share',
         params: { classId },
         query: { referrer },
       });
-      return;
+      return undefined;
     }
+    let nftMetadata;
     try {
       await Promise.all([
-        store.dispatch('fetchNFTMetadata', classId),
+        store.dispatch('fetchNFTClassMetadata', classId),
         store.dispatch('lazyGetNFTPurchaseInfo', classId).catch(err => {
           if (err.response?.data !== 'NFT_CLASS_NOT_FOUND') {
             // eslint-disable-next-line no-console
             console.error(JSON.stringify(err));
           }
         }),
       ]);
+      nftMetadata = await store.dispatch('fetchNFTMetadata', {
```

Comment:
Not sure if we should/can put this into the above `Promise.all`

Reply:
The original action depends on NFT Class metadata for fallback, I think we can resolve that with computed properties.